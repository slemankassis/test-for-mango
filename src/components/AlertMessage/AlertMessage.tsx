import React, { FC } from 'react';
import classnames from 'classnames';
import { makeStyles } from '@mui/styles';

// Version with Material UI instead of styled-components
const useStyles = makeStyles({
  alert: { color: '#fff', padding: '0.5em 0.5em 0.3em' },
  error: { backgroundColor: '#ff4444', boxShadow: '0 8px #cc0000' },
  info: {
    backgroundColor: '#33b5e5',
    boxShadow: '0 8px #0099cc',
  },
});

enum AlertMessageType {
  error = 'm-error',
  info = 'm-info',
}

export interface AlertMessageProps {
  type: keyof typeof AlertMessageType;
}

const AlertMessage: FC<AlertMessageProps> = ({ children, type }) => {
  const classes = useStyles();

  if (children) {
    return (
      <div className={classnames(alert, classes[type])}>{children}</div>
    );
  } else {
    return null;
  }
};

export default AlertMessage;
