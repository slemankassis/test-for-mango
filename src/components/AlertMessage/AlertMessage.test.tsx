import React from 'react';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import AlertMessage, { AlertMessageProps } from './AlertMessage';

const setup = (msgBoxProps: AlertMessageProps) => {
  const utils = render(<AlertMessage {...msgBoxProps}>test</AlertMessage>);

  return {
    el: utils.getByText('test'),
    ...utils,
  };
};

describe('AlertMessage check props', () => {
  test('should not exist when empty', () => {
    const { container } = render(<AlertMessage type="info"></AlertMessage>);

    expect(container.firstChild).toBe(null);
  });
});

describe('snapshots AlertMessage', () => {
  test('snapshot empty error match', () => {
    const tree = TestRenderer.create(<AlertMessage type="error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot empty info match', () => {
    const tree = TestRenderer.create(<AlertMessage type="info" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot full error match', () => {
    const tree = TestRenderer.create(
      <AlertMessage type="error">Test</AlertMessage>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot full error match', () => {
    const tree = TestRenderer.create(
      <AlertMessage type="info">Test</AlertMessage>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
