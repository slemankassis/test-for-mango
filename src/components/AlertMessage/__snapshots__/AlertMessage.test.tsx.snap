// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots AlertMessage snapshot empty error match 1`] = `null`;

exports[`snapshots AlertMessage snapshot empty info match 1`] = `null`;

exports[`snapshots AlertMessage snapshot full error match 1`] = `
<div
  className="makeStyles-error-5"
>
  Test
</div>
`;

exports[`snapshots AlertMessage snapshot full error match 2`] = `
<div
  className="makeStyles-info-6"
>
  Test
</div>
`;
